client/lib/abtest/active-tests.js

```diff
@@ -167,4 +167,14 @@ export default {
 		allowExistingUsers: true,
 		localeTargets: 'any',
 	},
+	reskinSignupFlow: {
```

Comment:
The scope of the test is:
> All non-EN users, including existing users, who come through onboarding flow

Could you clarify what is `onboarding` flow in this context (we are horrible with naming things). Is it only users coming from `/new`? Or users trying to create a new site?

Either way, it seems we only check whether they are coming from `onboarding` flow only in one place. However, the signup persists state (and we are including existing users) - meaning, we'll have users getting assigned to the `reskinned` variation who probably should not have been.

I'm also worried that the current implementation "taints" the whole signup flow - regardless where the user is coming from. Say, they got assigned to the `reskinned` variation (regardless why). They finish signup and then want to create a new site, but not from the `onboarding` flow. _At least_, some of the screens will still be showing the `reskinned` version, because they just check which variation the user is in.

....or I might be missing something obvious here, which is totally possible cause it's Sunday and this is the signup framework 😅

Reply:
That's a good point @klimeryk . Probably, the safest method would be to determine `reskinned` eligibility at one place and pass that down as props to the components depending on it. If we don't want to make this rewrite, I still think that we can cover our bases with a few extra checks, more on that below:

* The `isSignupStep` + `onboarding` [flow name check](https://github.com/Automattic/wp-calypso/blob/3ef43b2bdf6e27f176ab0b1ccea227d9c95c2795/client/signup/main.jsx#L696) ensures that if the domain UI is viewed outside of signup flow i.e in the site launch flow or domain managament page, then the reskinned variation will not be seen. 
* The plans step is missing a signup step check, we'll need to add one here.
* The user step is missing this check. So, if the user performs the following series of steps:
    - Click "Get Started" on https://wordpress.com logged out, and reach https://wordpress.com/start/user. 
    - At this point, they will be assigned to the reskin test. Assume they got assigned to the "reskinned" variant of the test.
    - User navigates away from the signup step without doing anythiing and goes back to wordpress.com to explore our landing pages more.
    - User then reaches https://wordpress.com/start/premium by clicking "Get Started" on the Premium plan. Then they will see the user step like this:

<img width="1332" alt="Screenshot 2020-08-04 at 4 44 43 PM" src="https://user-images.githubusercontent.com/1269602/89287547-d1ef7080-d671-11ea-992a-e56044d57712.png">

A quick way to repro this is to go to /start, manually assign yourself to the reskin test, and then go to /start/premium.

I think if we have the signup step checks in place, we should be good. But it's possible that I'm missing some scenarios, so please let me know if you think of any :)